src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -11,13 +11,19 @@ namespace Avalonia.Skia
     internal class DrawingContextImpl : IDrawingContextImpl
     {
         private Stack<PaintWrapper> maskStack = new Stack<PaintWrapper>();
-        
+        private Matrix _dpiScaleMatrix;
+        private Matrix _currentTransform = Matrix.Identity;
+
         public SKCanvas Canvas { get; private set; }
 
-        public DrawingContextImpl(SKCanvas canvas)
+        public DrawingContextImpl(SKCanvas canvas, double scale = 1.0)
         {
             Canvas = canvas;
             Canvas.Clear();
+
+            _dpiScaleMatrix = Matrix.CreateScale(scale, scale);
+
+            _currentTransform =  Matrix.Identity * _dpiScaleMatrix;
```

Comment:
You can think of `Matrix.Identity` as a 1 in a multiplication, so you don't need it here, as essentially you're saying `currentTranform = 1 * scale`. In other words ` Matrix.Identity * _dpiScaleMatrix == _dpiScaleMatrix`.